e2e/tests/interchain_accounts/groups_test.go

```diff
@@ -0,0 +1,214 @@
+		msgVote := &grouptypes.MsgVote{
+			ProposalId: InitialProposalID,
+			Voter:      chainAAddress,
+			Option:     grouptypes.VOTE_OPTION_YES,
+			Exec:       grouptypes.Exec_EXEC_TRY,
+		}
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgVote)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+	})
+
+	t.Run("start relayer", func(t *testing.T) {
+		s.StartRelayer(relayer)
+	})
+
+	t.Run("verify interchain account registration success", func(t *testing.T) {
+		interchainAccAddr, err = s.QueryInterchainAccount(ctx, chainA, groupPolicyAddr, ibctesting.FirstConnectionID)
+		s.Require().NoError(err)
+
+		channels, err := relayer.GetChannels(ctx, s.GetRelayerExecReporter(), chainA.Config().ChainID)
+		s.Require().NoError(err)
+		s.Require().Equal(len(channels), 2)
+	})
+
+	t.Run("fund interchain account wallet", func(t *testing.T) {
+		err := chainB.SendFunds(ctx, ibctest.FaucetAccountKeyName, ibc.WalletAmount{
+			Address: interchainAccAddr,
+			Amount:  testvalues.StartingTokenAmount,
+			Denom:   chainB.Config().Denom,
+		})
+		s.Require().NoError(err)
+	})
+
+	t.Run("submit proposal for MsgSendTx", func(t *testing.T) {
+		msgBankSend := &banktypes.MsgSend{
+			FromAddress: interchainAccAddr,
+			ToAddress:   chainBAddress,
+			Amount:      sdk.NewCoins(testvalues.DefaultTransferAmount(chainB.Config().Denom)),
+		}
+
+		cfg := simappparams.MakeTestEncodingConfig()
+		banktypes.RegisterInterfaces(cfg.InterfaceRegistry)
+		cdc := codec.NewProtoCodec(cfg.InterfaceRegistry)
+
+		bz, err := icatypes.SerializeCosmosTx(cdc, []sdk.Msg{msgBankSend})
+		s.Require().NoError(err)
+
+		packetData := icatypes.InterchainAccountPacketData{
+			Type: icatypes.EXECUTE_TX,
+			Data: bz,
+			Memo: "e2e",
+		}
+
+		msgSubmitTx := controllertypes.NewMsgSendTx(groupPolicyAddr, ibctesting.FirstConnectionID, uint64(time.Hour.Nanoseconds()), packetData)
+		msgSubmitProposal, err := grouptypes.NewMsgSubmitProposal(groupPolicyAddr, []string{chainAAddress}, []sdk.Msg{msgSubmitTx}, DefaultMetadata, grouptypes.Exec_EXEC_UNSPECIFIED)
+		s.Require().NoError(err)
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgSubmitProposal)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+	})
+
+	t.Run("vote and exec proposal", func(t *testing.T) {
+		msgVote := &grouptypes.MsgVote{
+			ProposalId: InitialProposalID + 1,
+			Voter:      chainAAddress,
+			Option:     grouptypes.VOTE_OPTION_YES,
+			Exec:       grouptypes.Exec_EXEC_TRY,
+		}
+
+		txResp, err := s.BroadcastMessages(ctx, chainA, chainAWallet, msgVote)
+		s.Require().NoError(err)
+		s.AssertValidTxResponse(txResp)
+	})
+
+	t.Run("verify tokens transferred", func(t *testing.T) {
+		s.Require().NoError(test.WaitForBlocks(ctx, 10, chainA, chainB), "failed to wait for blocks")
+
+		balance, err := chainB.GetBalance(ctx, chainBAddress, chainB.Config().Denom)
+		s.Require().NoError(err)
+
+		_, err = chainB.GetBalance(ctx, interchainAccAddr, chainB.Config().Denom)
```

Comment:
should you check that this balance has be subtracted as a sanity check?

Reply:
Done :)